### PR TITLE
Fixes #22642 - limit searching by host to explicit query

### PR DESCRIPTION
**Pull request by ares**

### CODE REVIEW

**Comment by ohadlevy** [comment]
isn't name is one of the most common queries? I would by default prefer to limit most of the other queries vs name? also, is name really adding a lot of complexity to the query to being with?

**Comment by ohadlevy** [comment]
oh - i take it back, i missread the pr, you are avoiding the searching on hosts name not hostgroup name :+1:

**Review by ohadlevy — APPROVED** [review]

**Review by tbrisker — CHANGES_REQUESTED** [review]
@ares Makes sense, but there's a bunch of other fields that i think should also be only explicit - lines 74-82 for sure, possibly also 58, and also might be good to add in katello extensions - https://github.com/Katello/katello/blob/master/app/models/katello/concerns/hostgroup_extensions.rb#L17

**Comment by ares** [comment]
@tbrisker I agree, fixed on lines you suggested too (except one which already had it), I'll send a PR to Katello too and link it to this one, good catch! could you please take another look?

File: app/models/hostgroup.rb [hunk 1]
```diff
@@ -53,7 +53,7 @@ class Hostgroup < ApplicationRecord
 
   scoped_search :on => :name, :complete_value => :true
   scoped_search :relation => :group_parameters,    :on => :value, :on_key=> :name, :complete_value => true, :only_explicit => true, :rename => :params
-  scoped_search :relation => :hosts, :on => :name, :complete_value => :true, :rename => "host"
+  scoped_search :relation => :hosts, :on => :name, :complete_value => :true, :rename => "host", :only_explicit => true
   scoped_search :relation => :puppetclasses, :on => :name, :complete_value => true, :rename => :class, :only_explicit => true, :operators => ['= ', '~ ']
   scoped_search :relation => :environment, :on => :name, :complete_value => :true, :rename => :environment
```
Comment (tbrisker) [inline app/models/hostgroup.rb]: What about this one? or is this intentional?

**Review by tbrisker — CHANGES_REQUESTED** [review]
I think you missed one :smile_cat:

File: app/models/hostgroup.rb (hunk 1 above)
Comment (ares) [reply app/models/hostgroup.rb]: no, but I only added to lines you suggested :-) added here too, mind to take one more look?

**Review by tbrisker — APPROVED** [review]
Thanks @ares ! Will merge once test pass.

**Pull request merged**